python/pysrc/kaskada/_timestream.py

```diff
@@ -111,10 +111,18 @@ def make_arg(arg: Union[Timestream, Literal]) -> _ffi.Expr:
 
         ffi_args = [make_arg(arg) for arg in args]
         try:
-            return Timestream(
-                # TODO: FRAZ - so I need a `call` that can take the udf
-                _ffi.Expr.call(session=session, operation=func, args=ffi_args)
-            )
+            if isinstance(func, str):
+                return Timestream(
+                    _ffi.Expr.call(session=session, operation=func, args=ffi_args)
+                )
+            elif isinstance(func, _ffi.Udf):
```

Comment:
Thoughts on whether this should be `_ffi.Udf` or `Udf`?

Reply:
I'm not sure I understand the question, or maybe the distinction well enough. the `_ffi.Udf` class is the rust-built class available in python that we're operating on in the `_ffi.Expr` class. 

Where `Udf` is the python class that wraps the `_ffi.Udf`, allows a decorator annotation, and defines how to execute the callable. The `__call__` function is defined as 
```
    def __call__(self, *args: Timestream | Literal) -> Timestream:
        """Apply the UDF to the given arguments."""
        return Timestream._call(self._ffi, *args)

```

which passes in the `self._ffi`. This is the `_ffi.Udf`. Are you asking here whether we should pass in `self` instead, then get out the `_ffi.Udf` within `TImestream._call`?